app/web/src/components/RecipeForm.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useEffect } from "react"
+import useRecipeState from "../hooks/useRecipeState.tsx";
+import { useParams, useHistory } from 'react-router-dom';
+import { addRecipe, getRecipe, updateRecipe } from "../context/RecipesService.tsx";
+import { RecipeActionTypes } from "../context/types.tsx";
+import { withRouter, RouteComponentProps } from "react-router-dom"
+import { RecipeContext } from "../context/RecipeContext.tsx";
+import styled from 'styled-components';
+import { StyledButton, RecipesListContainer, StyledLinkHeader, RecipesBookContainer, StyleTitle, StyledHeader } from './Styles.tsx';
+import { RecipeItemContainer, RecipeName, RecipeLabel } from "./Styles.tsx";
+
+const RecipesBookContainerAdd = styled(RecipesBookContainer)`
+    padding-bottom: 45px;
+`;
+const StyledButtonIngredient = styled(StyledButton)`
+    margin-top: 10px;
+    float: none;
+`;
+const StyledButtonIngredientDelete = styled(StyledButton)`
+    float: none;
+`;
+
+interface RecipeFormProps extends RouteComponentProps { }
+
+const RecipeForm: React.FC<RecipeFormProps> = (props) => {
+    const history = useHistory();
+    const { id } = useParams()
+    const { dispatch } = useContext(RecipeContext);
+
+    const {
+        recipe,
+        setRecipe,
+        setName,
+        setDescription,
+        addIngredient,
+        removeIngredient,
+        updateIngredient,
+        reset
+    } = useRecipeState();
+
+    useEffect(() => {
+        const fetchRecipe = async (id) => {
+            try {
+                let payload = await getRecipe(id)
+                setRecipe(payload);
+            } catch (error) {
+                dispatch({ type: RecipeActionTypes.ERROR, payload: "Something went wrong" });
+            }
+        };
+        if (id) {
+            fetchRecipe(id);
+        }
+    }, [dispatch, id]);
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        try {
+            let payload
+            if (props.actionType === RecipeActionTypes.ADD_RECIPE) {
+                payload = await addRecipe(recipe)
+            } else {
+                payload = await updateRecipe(recipe)
+            }
+            console.log('payload: ', payload);
+            dispatch({ type: props.actionType, payload: payload });
+            reset();
+            history.push("/");
+
+        } catch (error) {
+            dispatch({ type: RecipeActionTypes.ERROR, payload: "Something went wrong" });
+        }
+    }
+
+    return (
+        <RecipesBookContainerAdd>
+            <StyledHeader>
+                <StyleTitle>
+                    {props.actionType === RecipeActionTypes.ADD_RECIPE ? "Add Recipe" : "Edit Recipe"}
+                </StyleTitle>
+                <StyledLinkHeader to="/">Back to Home</StyledLinkHeader>
+            </StyledHeader>
+            <form onSubmit={(e) => handleSubmit(e)}>
```

Comment:
Since you're using `e.preventDefault()` to prevent the defautl behavior of the form, maybe there's no point in using a form and you can attach your `handleSubmit` to the Button's `onClick` instead. Does using a `form` add anything here?

Reply:
I think adding the form was an "old habit". I removed it and it does looks a little more cleaner.